ecmascript/dep-graph/src/lib.rs

```diff
@@ -0,0 +1,304 @@
+use swc_common::{
+    comments::{Comment, SingleThreadedComments},
+    Loc, DUMMY_SP,
+};
+pub use swc_common::{SourceMap, Span};
+use swc_ecma_ast as ast;
+use swc_ecma_visit::{self, Node, Visit, VisitWith};
+
+pub fn analyze_dependencies(
+    module: &ast::Module,
+    source_map: &SourceMap,
+    comments: &SingleThreadedComments,
+) -> Vec<DependencyDescriptor> {
+    let mut v = DependencyCollector {
+        comments,
+        source_map,
+        items: vec![],
+    };
+    module.visit_with(&ast::Invalid { span: DUMMY_SP }, &mut v);
+    v.items
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum DependencyKind {
+    Import,
+    ImportType,
+    Export,
+    ExportType,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct DependencyDescriptor {
+    pub kind: DependencyKind,
+    /// A flag indicating if the import is dynamic or not.
+    pub is_dynamic: bool,
+    /// Any leading comments associated with the dependency.  This is used for
+    /// further processing of supported pragma that impact the dependency.
+    pub leading_comments: Vec<Comment>,
+    /// The location of the import/export statement.
+    pub col: usize,
+    pub line: usize,
+    /// The text specifier associated with the import/export statement.
+    pub specifier: String,
```

Comment:
Is this type intentional?

I think it will not impact performance anyway, but just to be sure.

Reply:
Yeah, but I'm open to changing it, what's your suggestion?

Reply:
I think `JsWord`, which is interned version of `Box<str>`, would be better.
Same identifier is typically used widely across modules.